src/test/java/org/jreliability/bdd/AbstractBDDProviderTest.java

```diff
@@ -29,14 +31,17 @@
 	public void testGet() {
 		BDDProvider<String> provider = factory.getProvider();
 
-		provider.get("a");
-		provider.get("b");
+		BDD<String> a = provider.get("a");
+		BDD<String> b = provider.get("b");
+		Assert.assertNotEquals(a, b);
+		BDD<String> aPrime = provider.get("a");
+		Assert.assertEquals(a, aPrime);
 	}
 
 	/**
 	 * Tests the {@link getProvider} method with no arguments.
 	 */
-	@Test
+	@Test(expected = Test.None.class /* no exception expected */)
```

Comment:
Interesting.
* Should we throw an exception here?
* expected is only used for exceptions. If not an exception, we should use Assert.equals(Test.None.class, result.class)

Ignore this review if you have another opinion. Smartphone Review.

Reply:
Does this test tests anything? Either we should delete the whole test or really add tests to this method. Something like Assert.assertNotNull(a);
But even this test is more or less senseless. I mean - what purpose does it have? Why in a loop?